parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveComparator.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.schema;
+
+import org.apache.parquet.io.api.Binary;
+
+import java.nio.ByteBuffer;
+import java.util.Comparator;
+
+/**
+ * {@link Comparator} implementation that also supports the comparison of the related primitive type to avoid the
+ * performance penalty of boxing/unboxing. The {@code compare} methods for the not supported primitive types throw
+ * {@link UnsupportedOperationException}.
+ */
+public abstract class PrimitiveComparator<T> implements Comparator<T> {
+
+  public int compare(boolean b1, boolean b2) {
+    throw new UnsupportedOperationException("compare(boolean, boolean) was called on a non-boolean comparator");
```

Comment:
It would be helpful to include the comparator's class or simple name in these error messages.